app/kumactl/cmd/install/render_helm_files.go

```diff
@@ -0,0 +1,167 @@
+package install
+
+import (
+	"fmt"
+	"reflect"
+	"regexp"
+	"sort"
+	"strings"
+
+	"github.com/pkg/errors"
+	"helm.sh/helm/v3/pkg/chart"
+	"helm.sh/helm/v3/pkg/chart/loader"
+	"helm.sh/helm/v3/pkg/chartutil"
+	"helm.sh/helm/v3/pkg/engine"
+
+	"github.com/kumahq/kuma/app/kumactl/pkg/install/data"
+)
+
+var stripLabels = []string{
+	"app.kubernetes.io/managed-by",
+	"helm.sh/chart",
+	"app.kubernetes.io/version",
+}
+
+var kumaSystemNamespace = func(namespace string) string {
+	return fmt.Sprintf(`
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: %s
+  labels:
+    kuma.io/system-namespace: "true"
+`, namespace)
+}
+
+func renderHelmFiles(templates []data.File, args interface{}) ([]data.File, error) {
+	loadedChart, err := loadCharts(templates)
+	if err != nil {
+		return nil, errors.Errorf("Failed to load charts: %s", err)
+	}
+
+	overrideValues := generateOverrideValues(args)
+	if err := chartutil.ProcessDependencies(loadedChart, overrideValues); err != nil {
+		return nil, errors.Errorf("Failed to process dependencies: %s", err)
+	}
+
+	namespace := overrideValues["Namespace"].(string)
+	options := generateReleaseOptions(loadedChart.Metadata.Name, namespace)
+
+	valuesToRender, err := chartutil.ToRenderValues(loadedChart, overrideValues, options, nil)
+	if err != nil {
+		return nil, errors.Errorf("Failed to render values: %s", err)
+	}
+
+	files, err := engine.Render(loadedChart, valuesToRender)
+	if err != nil {
+		return nil, errors.Errorf("Failed to render templates: %s", err)
+	}
+	files["namespace.yaml"] = kumaSystemNamespace(namespace)
+
+	return postRender(loadedChart, files), nil
+}
+
+func loadCharts(templates []data.File) (*chart.Chart, error) {
+	files := []*loader.BufferedFile{}
+	for _, f := range templates {
+		files = append(files, &loader.BufferedFile{
+			Name: f.FullPath[1:],
+			Data: f.Data,
+		})
+	}
+
+	loadedChart, err := loader.LoadFiles(files)
+	if err != nil {
+		return nil, err
+	}
+
+	// Filter out the pre- templates
+	loadedTemplates := loadedChart.Templates
+	loadedChart.Templates = []*chart.File{}
+
+	for _, t := range loadedTemplates {
+		if !strings.HasPrefix(t.Name, "templates/pre-") {
+			loadedChart.Templates = append(loadedChart.Templates, &chart.File{
+				Name: t.Name,
+				Data: t.Data,
+			})
+		}
```

Comment:
This makes a lot of sense, though we might want to document the `pre-` convention for hooks. What about `post` hooks?